pkg/client/client.go

```diff
@@ -29,29 +39,30 @@ func CreateXrayClient(region string, datasourceInfo *awsds.AWSDatasourceSettings
 		config.Endpoint = aws.String(datasourceInfo.Endpoint)
 	}
 
-	clt := xray.New(sess, config)
-	clt.Handlers.Send.PushFront(func(r *request.Request) {
-		r.HTTPRequest.Header.Set("User-Agent", userAgentString())
-		log.DefaultLogger.Debug("CreateXrayClient", "userAgent", userAgentString())
-	})
-
-	return clt, nil
+	return xray.New(sess, config), nil
 }
 
 // CreateEc2Client creates client for EC2 api. We need this for some introspection queries like getting regions.
-func CreateEc2Client(region string, datasourceInfo *awsds.AWSDatasourceSettings) (*ec2.EC2, error) {
-	sess, err := sessions.GetSession(region, *datasourceInfo)
+func CreateEc2Client(awsSettings *awsds.AWSDatasourceSettings, backendSettings *backend.DataSourceInstanceSettings) (*ec2.EC2, error) {
+	httpClientProvider := httpclient.NewProvider()
+	httpClientOptions, err := backendSettings.HTTPClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	httpClient, err := httpClientProvider.New(httpClientOptions)
 	if err != nil {
+		backend.Logger.Error("failed to create HTTP client", "error", err.Error())
 		return nil, err
 	}
 
-	ec2Client := ec2.New(sess, &aws.Config{})
-	ec2Client.Handlers.Send.PushFront(func(r *request.Request) {
-		r.HTTPRequest.Header.Set("User-Agent", userAgentString())
+	sess, err := sessions.GetSession(awsds.SessionConfig{
+		Settings:      *awsSettings,
+		HTTPClient:    httpClient,
+		UserAgentName: aws.String("X-ray"),
 	})
```

Comment:
nit: you can save a few lines of code if you move this initialization to a function.